{"name": "发送文件-pdf", "status": "failed", "statusDetails": {"message": "AssertionError: 断言失败 -|- 实际结果:文件不存在 || 预期结果: 成功", "trace": "cls = <class 'tools.data_process.DataProcess'>\nresponse = {'errcode': '400004', 'errmsg': '文件不存在'}\nexpect_str = '{\"$.errmsg\":\"成功\"}'\n\n    @classmethod\n    def assert_result(cls, response: dict, expect_str: str):\n        \"\"\" 预期结果实际结果断言方法\n        :param response: 实际响应结果\n        :param expect_str: 预期响应内容，从excel中读取\n        return None\n        \"\"\"\n        # 后置sql变量转换\n        expect_str = rep_expr(expect_str, DataProcess.response_dict)\n        expect_dict = convert_json(expect_str)\n        index = 0\n        for k, v in expect_dict.items():\n            # 获取需要断言的实际结果部分\n            actual = extractor(response, k)\n            index += 1\n            logger.info(f'第{index}个断言,实际结果:{actual} | 预期结果:{v} \\n断言结果 {actual == v}')\n            allure_step(f'第{index}个断言',  f'实际结果:{actual} = 预期结果:{v}')\n            try:\n>               assert actual == v\nE               AssertionError\n\ntools/data_process.py:118: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\ncases = ['case_106', '消息中心应用消息', '发送消息', '发送文件-pdf', '/v1/message/send', '', ...]\n\n    @pytest.mark.flaky(reruns=0, reruns_delay=1)\n    def test_main(cases):#cases, get_db\n        # 此处的cases入参来自与 conftest.py  文件中 cases函数，与直接使用 @pytest.mark.parametrize\n        # 有着差不多的效果\n        # 发送请求\n        response, expect, sql = BaseRequest.send_request(cases)\n        # 执行sql\n        # DataProcess.handle_sql(sql, get_db)\n        # 断言操作\n>       DataProcess.assert_result(response, expect)\n\ntest/test_api.py:29: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ncls = <class 'tools.data_process.DataProcess'>\nresponse = {'errcode': '400004', 'errmsg': '文件不存在'}\nexpect_str = '{\"$.errmsg\":\"成功\"}'\n\n    @classmethod\n    def assert_result(cls, response: dict, expect_str: str):\n        \"\"\" 预期结果实际结果断言方法\n        :param response: 实际响应结果\n        :param expect_str: 预期响应内容，从excel中读取\n        return None\n        \"\"\"\n        # 后置sql变量转换\n        expect_str = rep_expr(expect_str, DataProcess.response_dict)\n        expect_dict = convert_json(expect_str)\n        index = 0\n        for k, v in expect_dict.items():\n            # 获取需要断言的实际结果部分\n            actual = extractor(response, k)\n            index += 1\n            logger.info(f'第{index}个断言,实际结果:{actual} | 预期结果:{v} \\n断言结果 {actual == v}')\n            allure_step(f'第{index}个断言',  f'实际结果:{actual} = 预期结果:{v}')\n            try:\n                assert actual == v\n            except AssertionError:\n>               raise AssertionError(f'断言失败 -|- 实际结果:{actual} || 预期结果: {v}')\nE               AssertionError: 断言失败 -|- 实际结果:文件不存在 || 预期结果: 成功\n\ntools/data_process.py:120: AssertionError"}, "steps": [{"name": "请求地址", "status": "passed", "attachments": [{"name": "请求地址", "source": "32a640d0-58cc-4dd0-b970-ed56e5a12c0c-attachment.txt", "type": "text/plain"}], "start": 1616132012509, "stop": 1616132012510}, {"name": "请求头", "status": "passed", "attachments": [{"name": "请求头", "source": "490d81ac-1c04-4254-be53-0c74141448dc-attachment.txt", "type": "text/plain"}], "start": 1616132012510, "stop": 1616132012510}, {"name": "请求参数", "status": "passed", "attachments": [{"name": "请求参数", "source": "886c8703-874d-490f-9a7a-dd801a4baeab-attachment.txt", "type": "text/plain"}], "start": 1616132012515, "stop": 1616132012516}, {"name": "上传文件", "status": "passed", "attachments": [{"name": "上传文件", "source": "b01f0013-93f7-4ceb-a969-5b32d98c4e71-attachment.txt", "type": "text/plain"}], "start": 1616132012516, "stop": 1616132012516}, {"name": "响应耗时(s)", "status": "passed", "attachments": [{"name": "响应耗时(s)", "source": "a77f8fbb-35a4-47fd-94b1-15b950f49dfb-attachment.txt", "type": "text/plain"}], "start": 1616132012622, "stop": 1616132012622}, {"name": "响应内容", "status": "passed", "attachments": [{"name": "响应内容", "source": "f4ff066e-0b9c-49c8-a3da-a0ad085aa3ea-attachment.txt", "type": "text/plain"}], "start": 1616132012622, "stop": 1616132012622}, {"name": "存储实际响应", "status": "passed", "attachments": [{"name": "存储实际响应", "source": "24c538fb-fa8a-467d-a92f-fe6378ef4cc1-attachment.txt", "type": "text/plain"}], "start": 1616132012623, "stop": 1616132012627}, {"name": "第1个断言", "status": "passed", "attachments": [{"name": "第1个断言", "source": "684b3415-7493-4a57-9169-45fd6168b40d-attachment.txt", "type": "text/plain"}], "start": 1616132012627, "stop": 1616132012627}], "parameters": [{"name": "cases", "value": "['case_106', '消息中心应用消息', '发送消息', '发送文件-pdf', '/v1/message/send', '', 'post', 'json', '', '{\\n \"appid\": \"&$.test_023..datas[8].app_id&\",\\n \"access_token\": \"&$.case_100.access_token&\",\\n \"to_users\": [\"&$.test_010..datas[0].uid&\", \"&$.test_011..datas[0].uid&\", \"&$.test_004..datas[0].uid&\", \"&$.test_005..datas[0].uid&\"],\\n \"type\": \"file\",\\n \"message\": {\\n  \"media_id\":\"&$.case_027.media_id&\"\\n },\\n \"msg_status\": \"1\"\\n}', '', '', '{\"$.errmsg\":\"成功\"}']"}], "start": 1616132012509, "stop": 1616132012627, "uuid": "aeeca753-1156-4bd7-8ecf-1e38700a8be0", "historyId": "567bbcf0b86c22c3b26dbdff47754ed9", "testCaseId": "881516e7ef2e2aa5436f6a3c3e6a5bcb", "fullName": "test.test_api#test_main", "labels": [{"name": "tag", "value": "flaky(reruns=0, reruns_delay=1)"}, {"name": "parentSuite", "value": "test"}, {"name": "suite", "value": "test_api"}, {"name": "host", "value": "liuchengxudeMacBook-Pro.local"}, {"name": "thread", "value": "2415-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test.test_api"}]}